### PR TITLE
New query stake pools command

### PR DESCRIPTION
**Pull request by newhoggy**
This PR depends on https://github.com/input-output-hk/cardano-node/pull/2977.

### CODE REVIEW

**Comment by newhoggy** [comment]
Example output:

```bash
$ CARDANO_NODE_SOCKET_PATH=socket cardano-cli query stake-pools --testnet-magic 1097911063 --cardano-mode
pool1qqyjr9pcrv97gwrueunug829fs5znw6p2wxft3fvqkgu5f4qlrg
pool1qqfnw2fwajdnam7xsqhhrje5cgd8jcltzfrx655rd23eqlxjfef
pool1qquwwu6680fr72y4779r2kpc7mxtch8rp2uhuqcc7v9p6q4f7ph
...
pool1lad5j5kawu60qljfqh02vnazxrahtaaj6cpaz4xeluw5xf023cg
pool1la4ghj4w4f8p4yk4qmx0qvqmzv6592ee9rs0vgla5w6lc2nc8w5
pool1llugtz5r4t6m7xz6es4qu7cszllm5y3uvx3ast5a9jzlv7h3xdu
```

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs [hunk 1]
```diff
@@ -794,7 +844,7 @@ executeQuery era cModeP localNodeConnInfo q = do
    execQuery :: IO (Either AcquireFailure (Either EraMismatch result))
    execQuery = queryNodeLocalState localNodeConnInfo Nothing q
 
-getSbe :: CardanoEraStyle era -> ExceptT ShelleyQueryCmdError IO (ShelleyBasedEra era)
+getSbe :: Monad m => CardanoEraStyle era -> ExceptT ShelleyQueryCmdError m (ShelleyBasedEra era)
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs:847]: Let's not make things polymorphic unless we have to.

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs [hunk 2]
```diff
@@ -374,6 +369,11 @@ runTxBuildRaw (AnyCardanoEra era)
     firstExceptT ShelleyTxCmdWriteFileError . newExceptT $
       writeFileTextEnvelope fpath Nothing txBody
 
+joinEither :: (x -> z) -> (y -> z) -> Either x (Either y a) -> Either z a
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:372]: Can we move these to the Helpers module in the cli?

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs [hunk 3]
```diff
@@ -440,38 +440,27 @@ runTxBuild (AnyCardanoEra era) (AnyConsensusModeParams cModeParams) networkId mS
...
+        newExceptT . joinEitherM ShelleyTxCmdAcquireFailure id $
+          executeLocalStateQueryExpr localNodeConnInfo Nothing $ \_ntcVersion -> runExceptT $ do
+            utxo <- firstExceptT ShelleyTxCmdTxSubmitErrorEraMismatch . newExceptT . queryExpr
+              $ QueryInEra eInMode $ QueryInShelleyBasedEra sbe
+              $ QueryUTxO (QueryUTxOByTxIn (Set.fromList onlyInputs))
+
+            pparams <- firstExceptT ShelleyTxCmdTxSubmitErrorEraMismatch . newExceptT . queryExpr
+              $ QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryProtocolParameters
 
-      if null txinsc
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:462]: You have removed txin checks here. Is this is mistake?

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**